### DISCUSSION TITLE
How do you convert CommonJS (CJS) to ES6?

### DISCUSSION THREAD

**Discussion by AsyncBanana** [babel/babel#11587] [General] [answered]
How do you convert Commonjs to ES6 modules? I want to convert it because of the numerous advantages (tree shaking, native performance, etc.) but the Babel docs only tell you how to convert it **from** ES6.

**Comment by nicolo-ribaudo** [comment, marked as answer]
Hi @awesomebanana2018!
We only provide plugins to compile your code from modern syntax to older one. However, there are many "unofficial" plugins that might help you.

What you are looking for is technically called a `codemod`: it's a transformation that you run on your source code only one time, and not as a build step to create a "dist" version of your app.

An example of such codemod/plugin is https://github.com/tbranyen/babel-plugin-transform-commonjs. However, please note that it might not fit exactly your needs, because a CommonJS->ES modules transformation could be done in many different ways. In that case, I suggest trying to write your own plugin that does exactly what you need.
